MaterialComponents.podspec

```diff
@@ -28,171 +28,195 @@ Pod::Spec.new do |mdc|
+  #.   component_name="#{component.base_name}"
   #
-  #    # Only if you have a resource bundle
-  #    component.resources = ["components/#{component.base_name}/src/Material#{component.base_name}.bundle"]
+  #    component.public_header_files = "components/#{component_name}/src/*.h"
+  #    component.source_files = "components/#{component_name}/src/*.{h,m}", "components/#{component_name}/src/private/*.{h,m}"
   #
+  #    # Only if you have a resource bundle
+  #    component.resources = ["components/#{component_name}/src/Material#{component_name}.bundle"]
   #  end
   #
 
-  mdc.subspec "ActivityIndicator" do |component|
-    component.subspec "Component" do |spec|
-      spec.ios.deployment_target = '8.0'
-      spec.public_header_files = "components/#{component.base_name}/src/*.h"
-      spec.source_files = "components/#{component.base_name}/src/*.{h,m}", "components/#{component.base_name}/src/private/*.{h,m}"
-      spec.resources = ["components/#{component.base_name}/src/Material#{component.base_name}.bundle"]
+  mdc.subspec "ActivityIndicator" do |spec|
+    component_name="#{spec.base_name}"
+
+    spec.subspec "Component" do |component|
+      component.ios.deployment_target = '8.0'
+      component.public_header_files = "components/#{component_name}/src/*.h"
+      component.source_files = "components/#{component_name}/src/*.{h,m}", "components/#{component_name}/src/private/*.{h,m}"
+      component.resources = ["components/#{component_name}/src/Material#{component_name}.bundle"]
 
-      spec.dependency "MDFInternationalization"
-      spec.dependency "MaterialComponents/Palettes"
-      spec.dependency "MaterialComponents/private/Application"
-      spec.dependency "MotionAnimator", "~> 2.0"
+      component.dependency "MDFInternationalization"
+      component.dependency "MaterialComponents/Palettes"
+      component.dependency "MaterialComponents/private/Application"
+      component.dependency "MotionAnimator", "~> 2.0"
     end
-    component.subspec "ColorThemer" do |spec|
-      spec.ios.deployment_target = '8.0'
-      spec.public_header_files = "components/#{component.base_name}/src/#{spec.base_name}/*.h"
-      spec.source_files = "components/#{component.base_name}/src/#{spec.base_name}/*.{h,m}"
-      spec.dependency "MaterialComponents/ActivityIndicator/Component"
-      spec.dependency "MaterialComponents/Themes"
+    spec.subspec "ColorThemer" do |colorthemer|
+      colorthemer.ios.deployment_target = '8.0'
+      colorthemer.public_header_files = "components/#{component_name}/src/ColorThemer/*.h"
+      colorthemer.source_files = "components/#{component_name}/src/ColorThemer/*.{h,m}"
+
+      colorthemer.dependency "MaterialComponents/#{component_name}/Component"
+      colorthemer.dependency "MaterialComponents/Themes"
     end
   end
 
   mdc.subspec "AnimationTiming" do |component|
+    component_name="#{component.base_name}"
```

Comment:
perhaps it would be less confusing if all our mdc.subspec's had a subspec?

The components that dont have themers/mutators would still get the same treatment in case they do get more utility build targets in the future.

Reply:
Ah, agreed. That'd help keep things consistent.